### PR TITLE
feat: added animation to menu button

### PR DESCRIPTION
**Pull request by iNavac03**
## 🔥 added icon animation to menu button 
----------------------------------------------

### Description
Added the icon animation to menu button on header, this animation added an icon that renders when hover on the link and increases opacity with time.

### How to test?
No tests needed

### Screenshots
Screenshot 2024-12-24 at 4 07 05 p m

#### After

### CODE REVIEW

File: apps/homepage/src/app/[lang]/_shared/menuButton/menuButton.tsx [hunk 1]
```diff
@@ -2,19 +2,34 @@
...
+  icon: UiIconProps["icon"];
 };
 
-export function MenuButton({ children, href }: menuButtonProps) {
+export function MenuButton({ children, href, icon }: menuButtonProps) {
+  const [showIcon, setShowIcon] = useState(false);
+
   return (
-    <Link href={href} className={styles.button}>
+    <Link href={href} className={styles.button} onMouseEnter={() => setShowIcon(true)} onMouseLeave={() => setShowIcon(false)}>
```
Comment (inavac182) [inline apps/homepage/src/app/[lang]/_shared/menuButton/menuButton.tsx]: Let's create a function outside the component render, something like:

```
const onMouseEnter = useCallback(() => {
  setShowIcon(true);
}, [setShowIcon]);
```

```
<Link href={href} className={styles.button} onMouseEnter={onMouseEnter} onMouseLeave={() => setShowIcon(false)}>
```

File: apps/homepage/src/app/_shared/menuButton/index.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export * from './menuButton'
```
Comment (inavac182) [inline apps/homepage/src/app/_shared/menuButton/index.ts]: What is this?

File: apps/homepage/src/app/_shared/header/header.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import styles from './header.module.scss'
```
Comment (inavac182) [inline apps/homepage/src/app/_shared/header/header.tsx]: Is this duplicated?

**Review by inavac182 — APPROVED** [review]

File: apps/homepage/src/app/[lang]/_shared/menuButton/menuButton.tsx [hunk 4]
```diff
@@ -2,19 +2,43 @@
...
+import { useCallback } from "react";
 
 type menuButtonProps = {
   children: React.ReactNode;
   href: string;
+  icon: UiIconProps["icon"];
 };
 
-export function MenuButton({ children, href }: menuButtonProps) {
+export function MenuButton({ children, href, icon }: menuButtonProps) {
```
Comment (inavac182) [inline apps/homepage/src/app/[lang]/_shared/menuButton/menuButton.tsx:15]: Next task is to move this button to the `ui` package.

**Review by adrife2005 — APPROVED** [review]
Good job !

**Pull request merged**